airflow/dags/gtfs_loader/validation_report_process.py

```diff
@@ -0,0 +1,72 @@
+# ---
+# python_callable: validator_process
+# provide_context: true
+# external_dependencies:
+#   - gtfs_downloader: validate_gcs_bucket
+# ---
+
+import pandas as pd
+import json
+
+from calitp import save_to_gcfs, read_gcfs
+
+PANDAS_TYPES_TO_BIGQUERY = {"O": "STRING", "i": "INTEGER", "f": "NUMERIC"}
+
+
+def process_notices(itp_id, url_number, validation):
+    raw_codes = validation["data"]["report"]["notices"]
+
+    # one row per code. includes a notices column with nested details
+    df_codes = pd.DataFrame(raw_codes)
+    df_codes.insert(0, "calitp_itp_id", itp_id)
+    df_codes.insert(1, "calitp_url_number", url_number)
+
+    return df_codes
+
+    return df_codes.explode("notices")
+
+
+def infer_notice_schema(notice_entries):
+    # note that convert_dtypes enables integer cols w/ missing values
+    wide = pd.DataFrame(notice_entries).convert_dtypes()
+    schema = []
+    for k in wide:
+        schema.append(dict(name=k, type=PANDAS_TYPES_TO_BIGQUERY[wide[k].dtype.kind]))
+
+    return schema
+
+
+def validator_process(execution_date, **kwargs):
+    base_path = f"schedule/{execution_date}"
+
+    status = pd.read_csv(read_gcfs(f"{base_path}/status.csv"))
+    success = status[lambda d: d.status == "success"]
+
+    # hold on to notices, so we can infer schema after
+    # note that I've commented out the code for inferring schema below,
+    # but it was usefule for generating, then hand-tweaking to load
+    # into bigquery
+    # notice_entries = []
+    for k, row in success.iterrows():
```

Comment:
we should be storing the validator runtime, schedule version next  to the validator rows? this might be doing it but not sure as I read through. 

ie. 

`gtfs_downloaded_at`: TS 
`gtfs_validated_by`: `validator_version`

Reply:
The validator version is stored in validation.json, but not in the data pulled out here. I can pull it in. The date the data was extracted will need to be added to everything for #69, and I'll use the strategy you list there to upsert validation as well.

`validation.json` looks like:

```
{
  version: "2.0.0", # added by gtfs-validator-api,
  data: {
    report: {
      # this is what we pull out
      ...
    },
    system_errors: {
      # I have never seen this non-empty, since I think it populates when the process fails
    },
  }
} 
```